### PR TITLE
storage: port sliders to PF4 slider component

### PR DESCRIPTION
**Pull request by KKoukiou**
Since storage was the last place where custom slider was used in this
repo the PF3 overrides were dropped as well.

With this PR:
-----------------

Screen Shot 2021-07-05 at 13 27 08
Screen Shot 2021-07-05 at 13 25 07

On master:
---------------
Screen Shot 2021-07-05 at 13 14 26
Screen Shot 2021-07-05 at 13 14 16

### CODE REVIEW

**Comment by martinpitt** [comment]
Nice! I found one small bug, though:

* Select the manual size input, and type a number:
 
image

 * Now press Tab to go to the unit field, press Cursor Down to move to GiB, or select it with the mouse
 * Press Tab again

Now the numeric value does not change:

image

and when I confirm, it really still considers "MB" as the actual unit. If OTOH I use the mouse to change the unit while the field is *not* focused, it converts correctly:

image

This is a regression, this does not happen on main.

**Comment by KKoukiou** [comment]
@martinpitt thanks for catching that issue, it's not a regression though, just reproduced exactly as you described on master. I will try to solve it within this PR however.

File: test/verify/check-storage-partitions [hunk 1]
```diff
@@ -95,6 +95,10 @@ class TestStoragePartitions(StorageCase):
         b.key_press(chr(37), use_ord=True)  # arrow left
         self.dialog_wait_val("size", 25.5)
 
+        # Check that changing units affects the text input
+        b.select_from_dropdown(".size-unit", "1073741824")
```
Comment (martinpitt) [inline test/verify/check-storage-partitions:99]: Confused.. The [firefox test fails](https://logs.cockpit-project.org/logs/pull-16046-20210706-094349-c82f3f97-fedora-34-firefox/log.html#284) on it not knowing what `chr(37)` is, but that code already existed!? It's breaking before this new line.

Comment (KKoukiou) [reply test/verify/check-storage-partitions:99]: Ehm this line you quote here (where is fails) is new and indeed the hardcoded keys for firefox did not include arrow left, reposted.

Comment (martinpitt) [reply test/verify/check-storage-partitions:99]: Right, but it failed on line 95 before, no? with the key_press(chr(37))

Comment (martinpitt) [reply test/verify/check-storage-partitions:99]: Ah, nevermind -- I only looked at the *second* patch, but the chr(37) is from the first. Thanks for fixing!

**Comment by martinpitt** [comment]
[this failure](https://logs.cockpit-project.org/logs/pull-16046-20210706-152757-566ae074-fedora-34-firefox/log.html#284) is interesting -- the test expects 25.5, but the [screenshot](https://logs.cockpit-project.org/logs/pull-16046-20210706-152757-566ae074-fedora-34-firefox/TestStoragePartitions-testSizeSlider-fedora-34-127.0.0.2-2201-FAIL.png) says 26. Perhaps relax the comparison to ≥ 25 and ≤ 26?

**Comment by KKoukiou** [comment]
> [this failure](https://logs.cockpit-project.org/logs/pull-16046-20210706-152757-566ae074-fedora-34-firefox/log.html#284) is interesting -- the test expects 25.5, but the [screenshot](https://logs.cockpit-project.org/logs/pull-16046-20210706-152757-566ae074-fedora-34-firefox/TestStoragePartitions-testSizeSlider-fedora-34-127.0.0.2-2201-FAIL.png) says 26. Perhaps relax the comparison to ≥ 25 and ≤ 26?

no, we are on 26, and I want to check that arrow left will decrease the value. It does not decrease it on firefox for some reason. As if arrow left does not take effect. I will need to take a look, but tomorow.

**Comment by KKoukiou** [comment]
> > [this failure](https://logs.cockpit-project.org/logs/pull-16046-20210706-152757-566ae074-fedora-34-firefox/log.html#284) is interesting -- the test expects 25.5, but the [screenshot](https://logs.cockpit-project.org/logs/pull-16046-20210706-152757-566ae074-fedora-34-firefox/TestStoragePartitions-testSizeSlider-fedora-34-127.0.0.2-2201-FAIL.png) says 26. Perhaps relax the comparison to ≥ 25 and ≤ 26?
> 
> no, we are on 26, and I want to check that arrow left will decrease the value. It does not decrease it on firefox for some reason. As if arrow left does not take effect. I will need to take a look, but tomorow.

Found it, it was my dummy mistake.

**Review by martinpitt — APPROVED** [review]

**Pull request merged**